File: help-the-bookseller/help-the-bookseller.py
import re
def stock_list(listOfArt, listOfCat):
    if  not listOfArt or not listOfCat:
        return ""
    t = []
    for i in listOfCat:
        s = 0
        for j in listOfArt:
            if j.startswith(i):
                s += int(re.search(r"\d+",j).group(0))
        t.append( "(" + i + " : "+str(s)+ ")")
    return " - ".join(t)

File: help-the-bookseller/help-the-bookseller.js
function stockList(listOfArt, listOfCat){
  if(listOfArt=="" || listOfCat==""){return ""}
  return listOfCat.map(a=>"("+a+" : "+counta(listOfArt,a)+")").join(" - ")
}

function counta(lista,st){
var rex = new RegExp("^"+st); 
return lista.reduce((a,b)=> rex.test(b)?a+parseInt(b.match(/\d+/g)):a,0)
}